src/pkg/clouds/aws/ecs/logs.go

```diff
@@ -284,3 +288,22 @@ func takeLastN[T any](input chan T, n int) chan T {
 	}()
 	return out
 }
+
+func takeFirstN[T any](input chan T, n int) chan T {
+	if n <= 0 {
+		return input
+	}
+	out := make(chan T)
+	go func() {
+		defer close(out)
+		count := 0
+		for evt := range input {
+			if count >= n {
+				break
+			}
+			out <- evt
+			count++
```

Comment:
`break` at the end, or we'd wait for the N+1th message before bailing:
```suggestion
			out <- evt
			count++
			if count >= n {
				break
			}
```